core/src/main/java/org/apache/calcite/sql/SqlOperator.java

```diff
@@ -539,7 +539,9 @@ public RelDataType inferReturnType(
       }
 
       // MEASURE wrapper should be removed, e.g. MEASURE<DOUBLE> should just be DOUBLE
-      if (isMeasure(returnType) && returnType.getMeasureElementType() != null) {
+      if (opBinding.getOperator().kind != SqlKind.AS
+          && isMeasure(returnType)
+          && returnType.getMeasureElementType() != null) {
         returnType = Objects.requireNonNull(returnType.getMeasureElementType());
```

Comment:
Is this still required given that we've just checked the element type isn't null?

Reply:
This your code! I'll remove the `requireNotNull` though.